pkg/provider/provider.go

```diff
@@ -308,37 +331,67 @@ func (p *Provider) GetSecretsStoreObjectContent(ctx context.Context, attrib, sec
 			return nil, err
 		}
 
-		// fetch the object from Key Vault
-		content, newObjectVersion, err := p.getKeyVaultObjectContent(ctx, kvClient, keyVaultObject, *vaultURL)
-		if err != nil {
-			return nil, err
-		}
+		// if version history isn't greater than 1, maintain the current behavior
+		if keyVaultObject.ObjectVersionHistory <= 1 {
```

Comment:
What if `ObjectVersionHistory > 1` but `ObjectVersion` is also explicitly pinned to a particular version of the  kv object?

Reply:
It returns up to `ObjectVersionHistory` versions of the secret starting at `ObjectVersion` (or the latest version if not specified)

Reply:
I added a comment to the relevant method to try indicating that a bit more clearly